### PR TITLE
#644: Implementation of Convolution on Graph

### PR DESCRIPTION
**Pull request by VIGS25**
Addressing #644, this PR adds in an implementation of the`KOrderGraphConv` and `AdaptiveFilter` layers mentioned in the paper.

They finally aggregate the features from different `KOrderGraphConv` layers and use them for graph tasks.

@rbharath: Would you also want an implementation of a model using these layers, for example using the `delaney` dataset?

### CODE REVIEW

File: contrib/hagcn/hagcn_layers.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+  def pow_k(inputs, k=1):
+    """Computes the kth power of inputs, used for adjacency matrix"""
+    if k == 1:
+      return inputs
+    if k == 0:
+      return tf.ones(inputs.shape)
+
+    if k % 2 == 0:
+      first_half = AdaptiveFilter.pow_k(inputs, int(k / 2))
+      second_half = AdaptiveFilter.pow_k(inputs, int(k / 2))
```
Comment (VIGS25) [inline contrib/hagcn/hagcn_layers.py]: Just realised that the `AdaptiveFilter.pow_k` method is more efficient with a single recursion for the `k % 2 == 0` case.
Will add in the next commit with a model to run the same.

**Comment by rbharath** [comment]
@VIGS25 This looks really cool! Yes, an implementation of a simple model using these layers would be great. With a couple of unit tests, this code would be a nice contribution to the DeepChem main trunk as well.

**Comment by rbharath** [comment]
@VIGS25 Looking good! Let me know when you think this one would be ready to merge in.

**Comment by VIGS25** [comment]
This is good to be merged in!
I'll add the unit tests later.

**Pull request merged**